song-server/src/main/java/bio/overture/song/server/service/ValidationService.java

```diff
@@ -98,59 +98,31 @@ public Optional<String> validate(@NonNull JsonNode payload) {
       if (analysisType.getOptions() != null && analysisType.getOptions().getFileTypes() != null) {
         fileTypes = analysisType.getOptions().getFileTypes();
       }
-
-      if(Objects.isNull(analysisType.getVersion())){
-        // newest version is fetched
-
+      // Only use the file type rules from the corresponding schema version
+      Integer schemaVersion = analysisType.getVersion();
+      if (Objects.isNull(schemaVersion)) {
+        // No version specified, use the latest schema version
         if (fileTypes.isEmpty()) {
-          // get file types from previous versions
-          List<AnalysisSchema> analysisSchemas =
-                  analysisTypeService.getAllAnalysisSchemas(analysisType.getName());
-
-          List<String> fileTypesFromPreviousVersions =
-                  analysisSchemas.stream()
-                          .filter(schema -> schema.getFileTypes() != null)
-                          .flatMap(schema -> schema.getFileTypes().stream())
-                          .distinct()
-                          .collect(Collectors.toList());
-
-          if (!fileTypesFromPreviousVersions.isEmpty()) {
-            validateFileType(fileTypes, payload);
+          // Reuse file types from the latest schema version
+          List<AnalysisSchema> previousSchemas = analysisTypeService.getAllAnalysisSchemas(analysisType.getName());
+          List<String> previousFileTypes = previousSchemas.stream()
+              .filter(schema -> schema.getFileTypes() != null)
+              .flatMap(schema -> schema.getFileTypes().stream())
+              .distinct()
+              .collect(Collectors.toList());
```

Comment:
Don't do this. Your comment above says "Only use the file type rules from the corresponding schema version" and now youre fetching other schema versions. Why?

The entire logic is just:

```java
if(!fileTypes.isEmpty()) { validateFileType(fileTypes, payload); }
```